radixdlt-java/src/main/java/com/radixdlt/client/application/translate/StatefulActionToParticlesMapper.java

```diff
@@ -0,0 +1,64 @@
+package com.radixdlt.client.application.translate;
+
+import com.radixdlt.client.atommodel.accounts.RadixAddress;
+import com.radixdlt.client.core.atoms.particles.SpunParticle;
+import io.reactivex.Observable;
+import java.util.Objects;
+
+/**
+ * Maps a high level application action to lower level spun particles used
+ * to construct an atom given a context requirement which this interface describes
+ * via requiredState().
+ */
+public interface StatefulActionToParticlesMapper {
+	class RequiredShardState {
+		private final Class<? extends ApplicationState> stateClass;
+		private final RadixAddress address;
+
+		public RequiredShardState(Class<? extends ApplicationState> stateClass, RadixAddress address) {
+			Objects.requireNonNull(stateClass);
+			Objects.requireNonNull(address);
+
+			this.stateClass = stateClass;
+			this.address = address;
+		}
+
+		/**
+		 * Retrieves the type of application state needed for this requirement
+		 *
+		 * @return the type of application state
+		 */
+		public Class<? extends ApplicationState> stateClass() {
```

Comment:
If feels a bit strange to have a class for this rather than an enum, interfaces or whatever.

Reply:
Basically this is the method by which this mapper can request for certain applicationState stores so it has to be something generic like a class or type.

Reply:
Given the need for constrained flexibility we could use something like a "state flavour" enum (i.e. type) you can request. Then there could be multiple implementations of that flavour.. kind of like https://docs.oracle.com/javase/tutorial/uiswing/dnd/dataflavor.html in clipboards where you can specify the flavor of data you can supply and handle. Not urgent by any means, just a thought for later.

Reply:
Not sure if we need that much flexibility, looks like flavor pattern uses Class + String when all we really need is Class?